src/project/flowr-analyzer-builder.ts

```diff
@@ -0,0 +1,37 @@
+import type { FlowrConfigOptions } from '../config';
+import { amendConfig, cloneConfig, defaultConfigOptions } from '../config';
+import type { DeepWritable } from 'ts-essentials';
+import type { RParseRequests } from '../r-bridge/retriever';
+import { FlowrAnalyzer } from './flowr-analyzer';
+import { retrieveEngineInstances } from '../engines';
+import type { KnownParser } from '../r-bridge/parser';
+
+export class FlowrAnalyzerBuilder {
+	private flowrConfig:      DeepWritable<FlowrConfigOptions> = cloneConfig(defaultConfigOptions);
+	private readonly request: RParseRequests;
+
+	public amendConfig(func: (config: DeepWritable<FlowrConfigOptions>) => FlowrConfigOptions) : FlowrAnalyzerBuilder {
```

Comment:
You can give `this` as a return type which harmonizes with builder extensions :)